amplify/src/main/java/com/github/stkent/amplify/tracking/GenericSettings.java

```diff
@@ -79,4 +79,8 @@ public T getEventValue(@NonNull final IEvent event) {
         return null;
     }
 
+    public boolean hasEventValue(@NonNull final IEvent event) {
+        return sharedPreferences.contains(event.getTrackingKey());
```

Comment:
This is potentially problematic if you track the same event in multiple ways e.g. with warmup and counter tracking... actually, the fact that the event tracking key is being used directly (without modification based on the type of the tracked value) seems problematic in and of itself?